src/Tools/dotnet-watch/src/DotNetWatcher.cs

```diff
@@ -91,10 +117,13 @@ public DotNetWatcher(IReporter reporter)
                         return;
                     }
 
+                    context.ChangedFile = fileSetTask.Result;
```

Comment:
Seems like we can have false negatives now if multiple files are changed at once.
e.g. Startup.cs and MyProj.csproj are both modified, the watcher returns Startup.cs as the file that changed (because it sees it first) and now we don't run restore when we should.

Reply:
You're right. It is using a vanilla FSW so it's entirely possible to miss updates too. We could use timestamps to make this consistent.

Reply:
I added a timestamp based check that should work for changes to (or removal of) proj files. Additions are a bit more problematic because we'd have to evaluate msbuild to know if new ones were added. I'm going to sit on that until we see actual reports of it. I know the current system has issues with additions, so it's not exacerbating the problem.